### PR TITLE
perf: collection/gallery explore  page images 

### PR DESCRIPTION
**Pull request by hassnian**
## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

First pr of more to come to close the main issue, this pr only focuses on the explore pages
mainly
- lazy loading of images
- use cfi size variants to load smaller images where possible 

### Collection Explore

- lazy load images: eagerly load first 3 rows of each fetch remaining are lazy loaded
     - first page load (image resources)
           - **mobile**: 2.5mb (this pr) vs 19.8mb (prod)
           - **desktop**: 3mb (this pr) vs 19.8mb (prod)
- use `metadata` from collection `meta` , saves an extra call, most collections didn't need that (ahp 87/94 collections), 
- use cfi size variants in collection images 
     - banner: about the same for me , but should fetch smaller images depending on the user's display
     - image: average of 8-10x smaller images  
- profile image: use cfi size variants
- remove: pwa.client.ts 

### Items Explore
- lazy load images: eagerly load first 2 rows of each fetch  remaining are lazy loaded

- [x] Ref #10416

## Needs QA check

- @kodadot/qa-guild please review

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16faLfsywwNATaEfbH2ah75dn6ZmctQWpMS5G4KFhbmj5hnD)

## Screenshot 📸

- [ ] My fix has changed **something** on UI;

### CODE REVIEW

**Comment by hassnian** [comment]
@preschian should we remove `providers/cloudflare.ts`, and route all images through the image worker ?

CleanShot 2024-06-15 at 10 11 00@2x

also some images fail to load like 

[cfi ❌](https://[kodadot.xyz/cdn-cgi/imagedelivery/jk5b6spi_m_-9qC4VTnjpg/bafybeiaq2lvwomxmtmqyr5yigi4mty7ybib4myqewv6c265aheqv6lvjgu/w=700)

[image worker ✅](https://[image-beta.w.kodadot.xyz/ipfs/bafybeiaq2lvwomxmtmqyr5yigi4mty7ybib4myqewv6c265aheqv6lvjgu)

**Comment by preschian** [comment]
> should we remove `providers/cloudflare.ts`, and route all images through the image worker ?

Yes, this one can help reduce the request counts. But we need to implement cfi size variants in image-worker first


> * use `metadata` from collection `meta` , saves an extra call, most collections didn't need that (ahp 87/94 collections),

if the `meta = null`, try to get the image from `metadata`

**Comment by hassnian** [comment]
> Yes, this one can help reduce the request counts. But we need to implement cfi size variants in image-worker first

updated fe , needs worker updated https://github.com/kodadot/workers/pull/313


> i'm having some problems when scrolling down images on ahp, at some point it takes a long time to be loaded(here i'm on page 3):
> 
> could be unrelated as we are having some problems on ahp

worked fine for me, @prury let's try again after https://github.com/kodadot/workers/pull/313 is merged

File: components/collection/CollectionCard.vue [hunk 1]
```diff
@@ -24,34 +27,46 @@
...
   image.value = sanitizeIpfsUrl(getCollectionImage(metadata) || '')
   isLoadingMeta.value = false
+}
+
+onMounted(async () => {
+  if (props.isLoading || !props.collection) {
+    return
+  }
+
+  const meta = props.collection.meta
```
Comment (preschian) [inline components/collection/CollectionCard.vue:63]: update types `Property meta does not exist on type`

File: plugins/pwa.client.ts
Comment (preschian) [inline plugins/pwa.client.ts:1]: on incognito, some assets still loaded through workbox
Screenshot 2024-06-24 at 13 27 50

File: components/collection/CollectionCard.vue (hunk 1 above)
Comment (hassnian) [reply components/collection/CollectionCard.vue:63]: done

**Review by preschian — APPROVED** [review]

**Comment by vikiival** [comment]
cc @Jarsen136 pls review
cc @hassnian pls resolve conflicts

**Review by Jarsen136 — APPROVED** [review]

**Comment by hassnian** [comment]
> cc @hassnian pls resolve conflicts

done

**Comment by hassnian** [comment]
> same here
> 
> > [vikiival](https://github.com/vikiival) added the [S-resolve-merge-conflicts-🤝](https://github.com/kodadot/nft-gallery/labels/S-resolve-merge-conflicts-%F0%9F%A4%9D) label [now](https://github.com/kodadot/nft-gallery/pull/10443#event-13553062471)

done

**Comment by vikiival** [comment]
One test is failing @hassnian

**Comment by hassnian** [comment]
> One test is failing @hassnian

@vikiival all tests are passing

**Pull request merged**